logstash-core/lib/logstash/instrument/metric_store.rb

```diff
@@ -51,12 +51,12 @@ def fetch_or_store(namespaces, key, default_value = nil)
       # BUT. If the value is not present in the `@fast_lookup` the value will be inserted and we assume that we don't
       # have it in the `@metric_store` for structured search so we add it there too.
 
-      value = @fast_lookup.get(namespaces.dup << key)
+      value = @fast_lookup.get((namespaces.dup << key).hash)
```

Comment:
@jordansissel 
> computing (namespaces.dup << key) twice seems odd.

Agreed, but it is pre-existing art. I assume it is because we can call 'get' many times per event (e.g. [m|b|tr]illions total), but on the put, we only call it for new metrics (e.g. dozens total).  I can change it, I don't see a reason why not. However, some diciplines prescribe limiting the scope of the change to ONLY that which is relevant to the problem. (e.g. keep the change set pure). I am unsure the guiding dicipline here.  

> Is namespaces a string? 

Nope, an array. The first thing I tried was to convert the key to a String. While it fixed the hot method in YourKit, it increased the wall time by over 10%. 

>  I wonder if we will look at this line of code and wonder why it was written this way. 

I tried to explained in the commit log, but I can add comment to the code too.  

@original-brownbear 
> Are you sure this is safe (enough)? 

I am never sure about any anything 😄 , but I believe this is safe enough. If I am reading this correctly it is creating a murmur hash from the hash values of the contents of the array. Since we only have a few dozen here i assume the risk of collision is very low. (is that your concern?)